### PR TITLE
HDDS-2286. Add a log info in ozone client and scm to print the exclus…

### PR DESCRIPTION
**Pull request by swagle**
##  What changes were proposed in this pull request?
Added additional logging to print exclude lists on client and SCM

(Please fill in changes proposed in this fix)

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-2286

## How was this patch tested?
Waiting on unit tests since only log statements were added.

### CODE REVIEW

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMBlockProtocolServer.java [hunk 1]
```diff
@@ -176,6 +176,10 @@ public void join() throws InterruptedException {
     auditMap.put("owner", owner);
     List<AllocatedBlock> blocks = new ArrayList<>(num);
     boolean auditSuccess = true;
+    LOG.info("Allocating blocks {} of size {}, with excludeList: " +
+        "datanodes = {}, pipelines = {}, containers = {}",
+        excludeList.getDatanodes(), excludeList.getPipelineIds(),
+        excludeList.getContainerIds());
```
Comment (adoroszlai) [inline hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMBlockProtocolServer.java]: There are 5 placeholders, but only 3 values.  Number and size of blocks seems to be missing.

Also I think `Allocating {} blocks` would be more natural than `Allocating blocks {}`.

Comment (swagle) [reply hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMBlockProtocolServer.java]: Thanks @adoroszlai for catching this.

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMBlockProtocolServer.java [hunk 2]
```diff
@@ -176,6 +176,10 @@ public void join() throws InterruptedException {
     auditMap.put("owner", owner);
     List<AllocatedBlock> blocks = new ArrayList<>(num);
     boolean auditSuccess = true;
+    LOG.info("Allocating blocks {} of size {}, with excludeList: " +
```
Comment (mukul1987) [inline hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMBlockProtocolServer.java]: Allocating blocks {} -> Allocating {} blocks might be better in terms of readability.

File: hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/io/BlockOutputStreamEntryPool.java [hunk 3]
```diff
@@ -254,6 +254,9 @@ long getKeyLength() {
    * @throws IOException
    */
   private void allocateNewBlock() throws IOException {
+    LOG.info("Allocating block with excludeList: datanodes = {}, " +
+        "pipelines = {}, containers = {}", excludeList.getDatanodes(),
+        excludeList.getPipelineIds(), excludeList.getContainerIds());
```
Comment (mukul1987) [inline hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/io/BlockOutputStreamEntryPool.java]: Lets add ExcludeList#toString and then lets use it at both the places.

**Pull request merged**